CharacterController.cs

```diff
@@ -0,0 +1,76 @@
+using Godot;
+using System;
+
+public class CharacterController : KinematicBody2D
+{
+    const float gravity = 200.0f;
+    const int walkSpeed = 200;
+    [Export]
+    public int JumpImpulse =  -400;
+    Vector2 velocity;
+    public AnimatedSprite _animatedSprite;
+    public string direction = "right";
+
+    public override void _Ready()
+    {
+        base._Ready();
+
+        _animatedSprite = GetNode<AnimatedSprite>("AnimatedSprite");
+    }
+
+    public override void _PhysicsProcess(float delta)
+    {
+        if (IsOnFloor())
+        {
+            velocity.y = gravity;
+        }
+        else
+        {
+            velocity.y += delta * gravity;
+        }
+		
+        if (Input.IsActionPressed("ui_left") && direction=="left")
```

Comment:
Instead of 
```
if (Input.IsActionPressed("ui_left") && direction=="left")
{
    velocity.x = -walkSpeed;
    _animatedSprite.Play("Run");
}
else if(Input.IsActionPressed("ui_left") && direction == "right")
{
    direction = "left";
    _animatedSprite.Scale = new Vector2(-.3f, .3f);
    velocity.x = -walkSpeed;
    _animatedSprite.Play("Run");
}
```


Try
```
if(Input.IsActionPressed("ui_left"))
{
    if(direction == "right")
    {
        direction = "left";
        _animatedSprite.Scale = new Vector2(-.3f, .3f);
    }
    velocity.x = -walkSpeed;
    _animatedSprite.Play("Run");
}
```